src/Sulu/Bundle/AdminBundle/Resources/js/components/InfiniteScroller/InfiniteScroller.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import React from 'react';
+import debounce from 'debounce';
+import type {Element, ElementRef} from 'react';
+
+const THRESHOLD = 100;
+
+type Props = {
+    children: Element<*>,
+    onLoad: (loadedPage: number) => void,
+    current: number,
+    total: number,
+};
+
+export default class InfiniteScroller extends React.PureComponent<Props> {
+    elementRef: ElementRef<'div'>;
+
+    scrollContainer: ElementRef<*>;
+
+    componentDidMount() {
+        this.scrollContainer = this.getScrollContainer(this.elementRef.parentNode);
+
+        this.bindScrollListener();
+    }
+
+    componentWillUnmount() {
+        this.unbindScrollListener();
+    }
+
+    componentDidUpdate() {
+        this.bindScrollListener();
+    }
+
+    getScrollContainer(parentContainer: ElementRef<*>) {
+        if (!parentContainer) {
+            return window.document.body;
+        }
+
+        if (this.isScrollAble(parentContainer)) {
+            return parentContainer;
+        }
+
+        return this.getScrollContainer(parentContainer.parentNode);
+    }
+
+    // We have to check for the overflow property inside the styling to detect if the container is scrollable
+    // otherwise (using scrollHeight) we would have issues with async content loads leading to wrong container sizes. 
+    isScrollAble(el: ElementRef<*>): boolean {
```

Comment:
I've actually written `isScrollable`, because `able` is not a separate word, and therefore shouldn't be written uppercase IMO.